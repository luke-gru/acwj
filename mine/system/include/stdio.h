#ifndef _STDIO_H_
#define _STDIO_H_

#include <stddef.h>
#include <stdarg.h>

#define EOF (-1)

// This FILE definition will do for now
typedef char *FILE;
extern FILE *stderr;
extern FILE *stdout;

FILE *fopen(char *pathname, char *mode);
size_t fread(void *ptr, size_t size, size_t nmemb, FILE *stream);
size_t fwrite(void *ptr, size_t size, size_t nmemb, FILE *stream);
int fclose(FILE *stream);
int printf(char *format);
int fprintf(FILE *stream, char *format);
size_t getline(char **lineptr, size_t *n, FILE *stream);

FILE *popen(const char *command, const char *type);
int pclose(FILE *stream);

int vfprintf(FILE *stream, const char *format, va_list ap);

#endif	// _STDIO_H_
