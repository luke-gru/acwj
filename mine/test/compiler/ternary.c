#include <stdio.h>

int x;
int y;

int main() {
  for (y= 0; y < 10; y++) {
    x = (y < 4) ? y :
      (y > 7) ? y+1000 : y + 9000;
    printf("%d\n", x);
  }

  return(0);
}
