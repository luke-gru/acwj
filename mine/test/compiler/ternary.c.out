0
1
2
3
9004
9005
9006
9007
1008
1009
